File: Python/0007.py
class Solution:
    def reverse(self, x: int) -> int:
        # Define the limits of a 32-bit signed integer
        INT_MIN = -2**31
        INT_MAX = 2**31 - 1
        
        # Initialize variables for storing the reversed number and handle negative sign
        reversed_x = 0
        is_negative = x < 0
        x = abs(x)
        
        # Reverse the digits of x
        while x != 0:
            # Extract the last digit
            digit = x % 10
            # Append it to reversed_x
            reversed_x = reversed_x * 10 + digit
            # Remove the last digit from x
            x //= 10
        
        # Apply the sign to reversed_x if the original number was negative
        if is_negative:
            reversed_x = -reversed_x
        
        # Check for integer overflow and return 0 if out of bounds
        if reversed_x < INT_MIN or reversed_x > INT_MAX:
            return 0
        
        return reversed_x

File: Ruby/0007.rb
def reverse(x)
    # Limits for 32-bit signed integer range
    int_min = -2**31
    int_max = 2**31 - 1
  
    # Handle negative numbers
    sign = x < 0 ? -1 : 1
    x = x.abs
  
    reversed = 0
    while x > 0
      pop = x % 10
      x /= 10
  
      # Check for overflow before updating reversed
      if reversed > (int_max - pop) / 10
        return 0
      end
  
      reversed = reversed * 10 + pop
    end
  
    # Restore the sign and check if the result is within the 32-bit integer range
    reversed *= sign
  
    if reversed < int_min || reversed > int_max
      return 0
    end
  
    reversed
  end
  